### PR TITLE
mui-datatables added missing types to MUIDataTableState

### LINKED ISSUE #45909: Outdated types on mui-datatables
Missing alot of types on [mui-datatables](https://www.npmjs.com/package/mui-datatables?activeTab=readme) package

https://github.com/gregnb/mui-datatables#options
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mui-datatables/index.d.ts#L162
for example : `confirmFilters` and `customFilterDialogFooter ` and the table state to with `sortOrder`

MUIDataTableState missing options versus normal tableState
image

Definitions by: Jeroen "Favna" Claassens @Favna  <https://github.com/favna>
//                 Ankith Konda @ankithkonda  <https://github.com/ankithkonda>
//                 Herman "Von" Waters IV @hwatersiv  <https://github.com/hwatersiv>
//                 souppower @souppower  <https://github.com/souppower>
//                 Byron "Byrekt" Mitchell @byrekt  <https://github.com/byrekt>

### PR DESCRIPTION
**Pull request by hugobarragon**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

closes #45909

### CODE REVIEW

File: types/mui-datatables/index.d.ts [hunk 1]
```diff
@@ -41,11 +42,14 @@ export interface MUIDataTableState {
     filterData: any[];
     filterList: string[][];
     page: number;
+    previousSelectedRow: null | { index: number; dataIndex: number };
     rowsPerPage: number;
     rowsPerPageOptions: number[];
     searchText: string | null;
+    searchProps: null | React.HTMLAttributes<HTMLInputElement>;
```
Comment (Favna) [inline types/mui-datatables/index.d.ts:49]:
```suggestion
    searchProps: React.HTMLAttributes<HTMLInputElement> | null;
```

Nit, but it is a convention to put `null` after the actual type.

**Review by favna — CHANGES_REQUESTED** [review]
Just one small change

**Comment by Favna** [comment]
Also can you add this to your original PR description?

```closes #45909```

That way your issue will auto close when this PR is merged.

**Review by favna — APPROVED** [review]

**Comment by hugobarragon** [comment]
if you need tests, tomorrow I can work them out

**Comment by Favna** [comment]
Nah it's fine, just undraft it.

**Comment by Favna** [comment]
> ❌ Only a DT maintainer can merge changes without tests

Well I guess if you can actually add tests tomorrow that'll speed up the merge. I didn't know that condition was in place.

File: types/mui-datatables/index.d.ts [hunk 2]
```diff
@@ -196,7 +200,7 @@ export interface MUIDataTableIsRowCheck {
         {
             index: number;
             dataIndex: number;
-        }
+        },
```
Comment (Favna) [inline types/mui-datatables/index.d.ts:203]:
```suggestion
        }
```

**Review by favna — CHANGES_REQUESTED** [review]
yes we got it CI, you don't like trailing commas

**Review by favna — APPROVED** [review]

**Comment by Favna** [comment]
Approved.

Taken from the bot:

> All of the items on the list are green. To merge, you need to post a comment including the string "Ready to merge" to bring in your changes.

**Comment by hugobarragon** [comment]
Ready to merge Jarvis

**Pull request merged**